packages/validator/src/util/externalSignerClient.ts

```diff
@@ -1,31 +1,199 @@
 import fetch from "cross-fetch";
+import {BeaconBlock, AggregateAndProof, AttestationData, VoluntaryExit, Fork} from "@lodestar/types/phase0";
+import {SyncCommitteeContribution} from "@lodestar/types/altair";
+import {ValidatorRegistrationV1} from "@lodestar/types/bellatrix";
+import {altair, Epoch, phase0, Root, Slot, ssz} from "@lodestar/types";
+import {toHexString} from "@chainsafe/ssz";
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+const serializerMap = {
+  ["AGGREGATION_SLOT"]: (data: Record<string, unknown>) => {
+    return data;
+  },
+  ["AGGREGATE_AND_PROOF"]: (data: Record<string, unknown>) => {
+    return ssz.phase0.AggregateAndProof.toJson(data as AggregateAndProof);
+  },
+  ["ATTESTATION"]: (data: Record<string, unknown>) => {
+    return ssz.phase0.AttestationData.toJson(data as AttestationData);
+  },
+  ["BLOCK_V2"]: (data: {version: string; block: BeaconBlock}) => {
+    if (data.version === "PHASE0") {
+      return {
+        version: data.version,
+        block: ssz.phase0.BeaconBlock.toJson(data.block as phase0.BeaconBlock),
+      };
+    } else if (data.version === "ALTAIR") {
+      return {
+        version: data.version,
+        block: ssz.altair.BeaconBlock.toJson(data.block as altair.BeaconBlock),
+      };
+    } else {
+      throw new Error(`version ${data.version} not supported by remote signer`);
```

Comment:
What about the bellatrix fork? Using this pattern doesn't scale and will be painful to mantain?

Reply:
I basically followed the pattern used in lighthouse [here](https://github.com/sigp/lighthouse/blob/aa022f46855df2a1420a6a80a788c73dc2779aa7/validator_client/src/signing_method/web3signer.rs#L76). Had an older version so bellatrix was not included. Added it now. As regards observation about maintenance. I'll revisit this after setting up all the tests

Reply:
Yet to do the final sanity check but it occurred to me, that maybe the maintenance won't be too hard. Because:

1. Having another fork is a coordinated effort that will be planned out and include coordinated code changes
2. Even if lodestar can handle new forks without code changes, it should not, because it also depends on making sure that the Web3Signer has been updated to also handle the new fork. If not, lodestar will pass the call to web3signer but it will fail.

Reply:
Right I wasnt aware of how weird the web3signer request types are 😅 made such that any fork after bellatrix with use the bellatrix path